src/tests/src/solver/modelParser/testModelTranslator.cpp

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright 2007-2024, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#define WIN32_LEAN_AND_MEAN
+
+#include <iostream>
+
+#include <boost/test/unit_test.hpp>
+
+#include "antares/solver/libObjectModel/library.h"
+#include "antares/solver/modelConverter/modelConverter.h"
+#include "antares/solver/modelParser/Library.h"
+
+using namespace Antares::Solver;
+
+namespace Antares::Solver::ObjectModel
+{
+inline std::ostream& operator<<(std::ostream& os, const ValueType& value_type)
+{
+    using namespace std::string_literals;
+    switch (value_type)
+    {
+    case ValueType::FLOAT:
+        os << "FLOAT"s;
+        break;
+    case ValueType::INTEGER:
+        os << "INTEGER"s;
+        break;
+    case ValueType::BOOL:
+        os << "BOOL"s;
+        break;
+    default:
+        os << "UNKNOWN"s;
+        break;
+    }
+    return os;
+}
+} // namespace Antares::Solver::ObjectModel
+
+// Test empty library
+BOOST_AUTO_TEST_CASE(test_library_empty)
+{
+    ModelParser::Library library;
+    ObjectModel::Library lib = ModelConverter::convert(library);
+    BOOST_CHECK(lib.id().empty());
+    BOOST_CHECK(lib.description().empty());
+    BOOST_CHECK(lib.portTypes().empty());
+    BOOST_CHECK(lib.models().empty());
+}
+
+// Test library with id and description
+BOOST_AUTO_TEST_CASE(test_library_id_and_description)
+{
+    ModelParser::Library library;
+    library.id = "test_id";
+    library.description = "test_description";
+    ObjectModel::Library lib = ModelConverter::convert(library);
+    BOOST_CHECK_EQUAL(lib.id(), "test_id");
+    BOOST_CHECK_EQUAL(lib.description(), "test_description");
+}
+
+// Test library with port types
+BOOST_AUTO_TEST_CASE(test_library_port_types_empty_fields)
+{
+    ModelParser::Library library;
+    ModelParser::PortType portType1{"port1", "flow port", {}};
+    ModelParser::PortType portType2{"port2", "impedance port", {}};
+    library.port_types = {portType1, portType2};
+    ObjectModel::Library lib = ModelConverter::convert(library);
```

Comment:
Here for instance, the **convert** (the action in the test) is lost among other steps (**arrange** and **assert**) and melted with them, so we have to look for it. 
Some empty lines around would be welcome.